pandas/core/computation/eval.py

```diff
@@ -207,13 +206,10 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
         scope. Most users will **not** need to change this parameter.
     target : a target object for assignment, optional, default is None
         essentially this is a passed in resolver
-    inplace : bool, default True
-        If expression mutates, whether to modify object inplace or return
-        copy with mutation.
-
-        WARNING: inplace=None currently falls back to to True, but
-        in a future version, will default to False.  Use inplace=True
-        explicitly rather than relying on the default.
+    inplace : bool, default False
+        If the expression mutates, whether to modify the expression inplace
```

Comment:
You changed object to expression in "whether to modify object inplace", but I think object is actually better.

Looking at the test example:
```
In [87]: df = DataFrame(np.random.randn(5, 2), columns=list('ab'))

In [88]: df.eval('c = a + b', inplace=False)
Out[88]: 
          a         b         c
0 -0.742101  2.084071  1.341970
1 -0.823597 -1.686791 -2.510388
2  0.277927 -0.076424  0.201503
3  1.159483  0.188500  1.347983
4  0.010328 -0.890899 -0.880571

In [89]: df
Out[89]: 
          a         b
0 -0.742101  2.084071
1 -0.823597 -1.686791
2  0.277927 -0.076424
3  1.159483  0.188500
4  0.010328 -0.890899
```

it is the `df` object that does or not get modified, not the "expression"